Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
+    @IBOutlet weak var showOrHideDetailsBtn: UIButton!
+
     var transactionHash: String?
     var transaction: Transaction!
-    
-    var generalTxDetails: [TransactionDetails] = []
-    
+
+    var generalTxDetails: [TransactionDetail] = []
+    var isTxInputsCollapsed: Bool = true
+    var isTxOutputsCollapsed: Bool = true
+    var isTxDetailsTableViewCollapsed: Bool = true
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        tableTransactionDetails
-            .hideEmptyAndExtraRows()
-            .autoResizeCell(estimatedHeight: 60.0)
-            .registerCellNib(TransactiontInputDetailsCell.self)
-        
-        tableTransactionDetails.registerCellNib(TransactionDetailCell.self)
-        tableTransactionDetails.registerCellNib(TransactiontInputDetailsCell.self)
-        tableTransactionDetails.registerCellNib(TransactiontOutputDetailsCell.self)
-    }
-    
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        self.navigationController?.navigationBar.isHidden = false
-
-        self.navigationItem.title = LocalizedStrings.transactionDetails
-        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "left-arrow"),
-                                                                style: .done, target: self,
-                                                                action: #selector(back))
-       
-        let optionsMenuButton = UIButton(type: .custom)
-        optionsMenuButton.setImage(UIImage(named: "right-menu"), for: .normal)
-        optionsMenuButton.addTarget(self, action: #selector(showOptionsMenu), for: .touchUpInside)
-        optionsMenuButton.frame = CGRect(x: 0, y: 0, width: 10, height: 51)
-        self.navigationItem.rightBarButtonItems = [
-            UIBarButtonItem(customView: optionsMenuButton)
-        ]
-        
+
+        self.showOrHideDetailsBtn.addBorder(atPosition: .top, color: UIColor.appColors.gray, thickness: 1)
+
         if self.transaction == nil && self.transactionHash != nil {
             let txHash = Data(fromHexEncodedString: self.transactionHash!)!
             var getTxError: NSError?
             let txJsonString = WalletLoader.shared.firstWallet?.getTransaction(txHash, error: &getTxError)
             if getTxError != nil {
                 print("wallet.getTransaction error", getTxError!.localizedDescription)
             }
-            
+
             do {
-                self.transaction = try JSONDecoder().decode(Transaction.self, from:(txJsonString!.utf8Bits))
+                self.transaction = try JSONDecoder().decode(Transaction.self, from: (txJsonString!.utf8Bits))
             } catch let error {
                 print("decode transaction error:", error.localizedDescription)
             }
         }
-        
-        self.prepareTransactionDetails()
+
+        self.prepareGeneralTxDetails()
+        self.displayTitle()
```

Comment:
I'd say `displayTitle` before `prepareGeneralTxDetails`. The actual function bodies will need to be re-ordered as well.